src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -1,50 +1,149 @@
 package codeu.controller;
 
 import java.io.IOException;
+import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
 
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+
 public class AdminServlet extends HttpServlet {
 
-    private HashSet<String> admins;
+  private HashSet<String> admins;
+
+  private UserStore userStore;
+  private ConversationStore conversationStore;
+  private MessageStore messageStore;
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+
+    setAdminUsernames();
+
+    setUserStore(UserStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+  }
+
+  void setAdminUsernames() {
+    admins = new HashSet<String>();
+    admins.add("admin");
+    admins.add("daniel");
+    admins.add("leslie");
+    admins.add("serena");
+    admins.add("shana");
+    admins.add("kyra");
+  }
+
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
 
-    @Override
-    public void init() throws ServletException {
-        super.init();
+  void computeAdminStats(HttpServletRequest request) {
 
-        setAdminUsernames();
+    // Calculate information to be displayed
+
+    // Users
+    List<User> allUsers = userStore.getAllUsers();
+
+    String newestUser = "N/A";
+    String mostActiveUser = "N/A";
+
+    // get newest user
+    if (allUsers.size() > 0) {
+      newestUser = allUsers.get(allUsers.size() - 1).getName();
```

Comment:
The User object has a creationTime field: https://github.com/Sereezy/CodeU-Spring-2018/blob/master/src/main/java/codeu/model/data/User.java#L58

Is getting the last user in the list guaranteed to always agree with those time stamps? I'm especially curious if we restart the server on AppEngine and load the users from the datastore. If we want this to be guaranteed, we should document it in the UserStore method.

Since this PR is pretty big already, if you decide this needs more work you could consider either removing the newest user field for now, or else adding a TODO comment that says we should investigate this further.